src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -243,6 +246,13 @@ public Builder identifyAs(String serviceName, String datacenter, String hostname
 			return this;
 		}
 
+		public Builder withTag(String tag) {
+			if ( tag != null) {
```

Comment:
Yes, in fact, what's even more correct: `Preconditions.checkArgument(!Strings.isNullOrEmpty(tag), "enter a non-empty tag name")